signer/src/transaction_signer.rs

```diff
@@ -182,6 +183,16 @@ where
 
     #[tracing::instrument(skip_all, fields(chain_tip = tracing::field::Empty))]
     async fn handle_signer_message(&mut self, msg: &network::Msg) -> Result<(), Error> {
+        if matches!(
+            msg.payload,
+            message::Payload::SignerDepositDecision(_)
+                | message::Payload::SignerWithdrawalDecision(_)
+                | message::Payload::StacksTransactionSignature(_)
+                | message::Payload::BitcoinTransactionSignAck(_)
+        ) {
```

Comment:
This one isn't so bad, but I find this style harder to read than the two step approach:
```rust
// The tx-signer does not handle these message types at all, the first two are
// handled by the request-decider (we should change to tx-checker or something)
// and the last two by the tx-coordinator.
let is_ignored_message = matches!(
    msg.payload,
    message::Payload::SignerDepositDecision(_)
    | message::Payload::SignerWithdrawalDecision(_)
    | message::Payload::StacksTransactionSignature(_)
    | message::Payload::BitcoinTransactionSignAck(_)
);

if is_ignored_message {
    return Ok(());
}
```